### PR TITLE
Use mapzen's autocomplete service

### PR DESCRIPTION
**Pull request by evansiroky**
also, cache autocomplete results so queries to Mapzen don’t occur when
user hits backspace

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -88,16 +91,26 @@ class Geocoder extends Component {
...
     } else {
-      search({
+      const autocompleteQuery = {
         apiKey,
         boundary,
         focusPoint,
         text: input
-      }).then((geojson) => {
+      }
+      const autocompleteQueryKey = stableStringify(autocompleteQuery)
```
Comment (trevorgerhardt) [inline index.js:100]: Curious, have you found `JSON.stringify` to be unstable?

**Review by trevorgerhardt — APPROVED** [review]
👍

File: index.js (hunk 1 above)
Comment (evansiroky) [reply index.js:100]: The reason I'm using stable stringify is that the following produce 2 different results:

```js
console.log(JSON.stringify({a:1,b:2}))  // "{"a":1, "b":2}"
console.log(JSON.stringify({b:1,a:2}))  // "{"b":2, "a":1}"
```

Comment (trevorgerhardt) [reply index.js:100]: I know that but it's a contrived example. I was wondering if you actually found it to be unstable in practice.

**Pull request merged**